File: analytics/src/main/java/io/appmetrica/analytics/impl/preparer/StringValueComposer.java
package io.appmetrica.analytics.impl.preparer;

import android.text.TextUtils;
import androidx.annotation.NonNull;
import io.appmetrica.analytics.coreutils.internal.StringUtils;
import io.appmetrica.analytics.impl.request.ReportRequestConfig;

public class StringValueComposer implements ValueComposer {

    @NonNull
    @Override
    public byte[] getValue(@NonNull EventFromDbModel event, @NonNull ReportRequestConfig config) {
        if (TextUtils.isEmpty(event.getValue()) == false) {
            return StringUtils.getUTF8Bytes(event.getValue());
        }
        return new byte[0];
    }
}

File: analytics/src/main/java/io/appmetrica/analytics/impl/preparer/StringValueComposer.kt
package io.appmetrica.analytics.impl.preparer

import io.appmetrica.analytics.coreutils.internal.StringUtils
import io.appmetrica.analytics.impl.request.ReportRequestConfig

internal class StringValueComposer : ValueComposer {

    override fun getValue(event: EventFromDbModel, config: ReportRequestConfig): ByteArray {
        return event.foldValue(
            onLegacy = StringUtils::getUTF8Bytes,
            onBytes = { it },
            onEmpty = { ByteArray(0) },
        )
    }
}
